### PR TITLE
added image-registry check

### PR DESCRIPTION
**Pull request by shashank-boyapally**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Added image registry check for kube-burner-ocp, this can be used for bare-metal instances where image-registry co is not present by default.

## Related Tickets & Documents

- Related Issue #79
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by rsevilla87** [comment]
Isn't the registry image registry status already verified by the cluster operators check?

https://github.com/kube-burner/kube-burner-ocp/blob/4a36d42712bd22e9308035149a3de93b33896f54/cluster_health.go#L67-L75

🤔

**Comment by rsevilla87** [comment]
I think this PR won't be necessary, IUC, Anil wasn't passing the `--check-health`  flag to the kube-burner-ocp command

**Comment by rsevilla87** [comment]
@shashank-boyapally , maybe we should default to true the value of this flag?, wdyt @jtaleric @afcollins @venkataanil ?

**Comment by shashank-boyapally** [comment]
I was thinking since we get the list of installed `operators`, we cannot check the condition of `image-registry` if it is not present, hence I added an additional check, if I'm not mistaken image-registry is not present when we spin up a bm instance, please correct me if I'm wrong. Yes, we can enable the flag to default true.

**Comment by shashank-boyapally** [comment]
Looks like the image-registry is present, hence we might not need explicit check to image-registry. Please confirm if I can close this PR as not needed anymore.

**Comment by venkataanil** [comment]
Its better we get a BM deployment and reporduce this issue and then find root cause and fix it.

**Comment by afcollins** [comment]
> find root cause
@venkataanil We know it already. Bare metal deployments do not include the image-registry by default.

It is [documented](https://docs.openshift.com/container-platform/4.16/registry/configuring_registry_storage/configuring-registry-storage-baremetal.html#registry-removed_configuring-registry-storage-baremetal) as a Day 2 operation.

Configuring a registry is not something kube-burner would do.

File: cluster-density.go [hunk 1]
```diff
@@ -52,6 +53,11 @@ func NewClusterDensity(wh *workloads.WorkloadHelper, variant string) *cobra.Comm
 			os.Setenv("INGRESS_DOMAIN", ingressDomain)
 		},
 		Run: func(cmd *cobra.Command, args []string) {
+			kubeClientProvider := config.NewKubeClientProvider("", "")
```
Comment (rsevilla87) [inline cluster-density.go:56]: Let's use this check only for the `cluster-density-v2` variant

File: cluster_health.go [hunk 2]
```diff
@@ -92,3 +92,16 @@ func isClusterHealthy(clientset kubernetes.Interface, openshiftClientset *versio
...
 }
+
+func clusterImageRegistryCheck(clientset kubernetes.Interface) bool {
+	deployment, err := clientset.AppsV1().Deployments("openshift-image-registry").Get(context.TODO(), "image-registry", metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("Error getting deployment: %v", err)
+		return false
+	}
+	if deployment.Status.AvailableReplicas > 0 {
+		log.Infof("Deployment image-regsitry in namespace openshift-image-registry is available with %d replicas", deployment.Status.AvailableReplicas)
```
Comment (afcollins) [inline cluster_health.go:103]: Is this an Info or Debug level message?

Comment (shashank-boyapally) [reply cluster_health.go:103]: I can move it to Debug level if we consider as debug..

Comment (afcollins) [reply cluster_health.go:103]: I think it is at most a debug

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**